src/client/api/types.ts

```diff
@@ -94,6 +102,10 @@ export type PythonApi = {
      * @param resource file that determines which connection to return
      */
     getLanguageServer(resource?: InterpreterUri): Promise<ILanguageServer | undefined>;
+    /**
+     * Registers a visibility filter for the interpreter status bar.
+     */
+    registerInterpreterStatusFilter(filter: IInterpreterStatusbarVisibilityFilter): void;
```

Comment:
It's a bit funny to me that the name of the function doesn't line up with the name of the thing being registered. ie. InterpreterStatusFilter doesn't sound like the same thing as InterpreterStatusbarVisibilityFilter. But I must admit that registerInterpreterStatusbarVisibilityFilter is a bit of a mouthful, and it's clear from the parameter types. So ok with leaving as is.